### PR TITLE
Overhaul article-body-adverts and extricate mobile from desktop article ads.

### PR DESCRIPTION
**Pull request by JonNorman**
## What does this change?
This PR aims to simplify the logic in article-body-adverts, the module responsible for inserting inline adverts into articles.

There are quite a lot of business logic variations and caveats in this file and, whilst this PR increases the number of lines of code by replicating some common code, I think this is to the good and allows for an easier reading of what's going on.

The main changes are:

- the AB test to change the rules has been removed during this refactor. It can be added back during a subsequent PR.

- the rules for mobile and desktop inline ads have been separated.

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/article-body-adverts.js [hunk 1]
```diff
@@ -149,78 +123,99 @@ const getRules = (): Object => {
...
-            .then(addInlineAds)
-            .then(stop);
-
-        return im;
+        return Promise.resolve();
     }
 
-    addInlineAds().then(stop);
-    return Promise.resolve(true);
+    const im = config.page.hasInlineMerchandise
```
Comment (katebee) [inline static/src/javascripts/projects/commercial/modules/article-body-adverts.js]: Could use `config.get(...)` for safer accessing? However, appreciate that this is code from before

Comment (JonNorman) [reply static/src/javascripts/projects/commercial/modules/article-body-adverts.js]: Yep, good shout - I can update that 👍 

Need to fix the tests too 😩

File: static/src/javascripts/projects/commercial/modules/article-body-adverts.js [hunk 2]
```diff
@@ -149,78 +123,99 @@ const getRules = (): Object => {
...
-        // we must wait for DFP to return, since if the merch
-        // component is empty, it might completely change the
-        // positions where we insert those MPUs.
-        im
-            .then(waitForMerch)
-            .then(addInlineAds)
-            .then(stop);
-
-        return im;
+        return Promise.resolve();
```
Comment (katebee) [inline static/src/javascripts/projects/commercial/modules/article-body-adverts.js]: I think we discussed the JS tests and use of `false` to help with testing? I agree that the module shouldn't have to bend to testing.

Could the  `expect(executionResult).toBe(false);` line be taken out of the spec? It seems that test will still work just by relying on the Spacefiller mock. That will allow you to remove the false here and not break the test :)

**Comment by katebee** [comment]
This looks good, and makes sense. With some of the logic moving around I expect the JS tests will be broken 😿 

Let me know if you are free to talk me through the changes tomorrow 😁

File: static/src/javascripts/projects/commercial/modules/article-body-adverts.js (hunk 2 above)
Comment (JonNorman) [reply static/src/javascripts/projects/commercial/modules/article-body-adverts.js]: Great idea, I'll get on it.

**Review by katebee — APPROVED** [review]
👍 Looks good

**Pull request merged**